lib/fog/openstack/models/image_v2/images.rb

```diff
@@ -18,9 +18,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-                image_hash.merge(:service => service))
+            result = all.find { |image| image.id == id }
```

Comment:
@Carthaca @dhague If we go back to the original implementation (`service.get_image_by_id(id).body`) instead of `all.find`, we get the following error when calling `reload`:

```
Fog::Image::OpenStack#test_0004_Reloads image state from the server:
NoMethodError: undefined method `get' for nil:NilClass
    /Library/Ruby/Gems/2.0.0/gems/fog-core-1.42.0/lib/fog/core/model.rb:43:in `reload'
    /Users/pivotal/workspace/fog-openstack/spec/image_v2_spec.rb:150:in `block (3 levels) in <top (required)>'
    /Library/Ruby/Gems/2.0.0/gems/vcr-3.0.3/lib/vcr/util/variable_args_block_caller.rb:9:in `call'
    /Library/Ruby/Gems/2.0.0/gems/vcr-3.0.3/lib/vcr/util/variable_args_block_caller.rb:9:in `call_block'
    /Library/Ruby/Gems/2.0.0/gems/vcr-3.0.3/lib/vcr.rb:189:in `use_cassette'
    /Users/pivotal/workspace/fog-openstack/spec/image_v2_spec.rb:125:in `block (2 levels) in <top (required)>'
```

This nil pointer is due to the base model having a nil `collection`: https://github.com/fog/fog-core/blob/f6ef6c3d9458bc235be57100acdced264e85e52a/lib/fog/core/model.rb#L43.

Any suggestions for an implementation that only fetches a single object from the server, but avoids this nil pointer exception?